kili/services/label_import/parser/__init__.py

```diff
@@ -0,0 +1,94 @@
+"""
+Parser classes
+"""
+import csv
+import json
+from abc import abstractmethod
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Tuple
+
+from kili.services.label_import.types import Classes
+
+
+class AbstractLabelParser:  # pylint: disable=too-few-public-methods
+    """
+    Abstract label parser
+    """
+
+    def __init__(self, classes_by_id: Optional[Classes], target_job: Optional[str]) -> None:
+        self.classes_by_id = classes_by_id
+        self.target_job = target_job
+
+    @abstractmethod
+    def parse(self, label_file: Path) -> Dict[str, Any]:
+        """
+        Parses a label file
+        """
+
+
+class YoloLabelParser(AbstractLabelParser):  # pylint: disable=too-few-public-methods
+    """
+    Yolo label parser
+    """
+
+    def parse(self, label_file: Path) -> Dict[str, Any]:
+        """
+        Parses a Yolo label file
+        """
+        annotations = []
+        assert self.classes_by_id
+        assert self.target_job
+        with open(label_file, "r", encoding="ascii") as l_f:
+            csv_reader = csv.reader(l_f, delimiter=" ")
+            for row in csv_reader:
+                vertices, category, proba = self._parse(row)
+                annotations.append(
+                    {
+                        "boundingPoly": [
+                            {"normalizedVertices": [{"x": v[0], "y": v[1]} for v in vertices]}
+                        ],
+                        "categories": [
+                            {
+                                "name": self.classes_by_id[category],
+                                "confidence": 100 if proba is None else int(100 * float(proba)),
+                            }
+                        ],
+                    }
+                )
+
+        kili_json_response = {self.target_job: {"annotations": annotations}}
+
+        return kili_json_response
+
+    @staticmethod
+    def _parse(row) -> Tuple[List[List[float]], int, Optional[float]]:
+        try:
+            c, x, y, w, h, p = row  # pylint: disable=invalid-name
```

Comment:
could we get more detailed name ? class_id, x, y, width, height ?

Reply:
done